### PR TITLE
feat(checkbox): add basic styling and CSS vars for new syntax

### PR DESCRIPTION
**Pull request by amandaejohnston**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Checkboxes using the new syntax aren't properly rendered or styled.

Issue URL: N/A

## What is the new behavior?

- General markup and styling added for new syntax, using a new `legacy-checkbox` class to section off styles from the old syntax.
- `--background` and `--background-checked` CSS properties renamed to `--checkbox-background` and `--checkbox-background-checked` respectively to avoid confusion with similar variables on `ion-item`.
- `basic` and `indeterminate` tests added back in for new syntax.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

The `--background` and `--background-checked` CSS properties have been renamed to `--checkbox-background` and `--checkbox-background-checked` respectively.

### CODE REVIEW

**Comment by amandaejohnston** [comment]
There are some errant datetime screenshots in here still, but I'm putting this up for review now to get the relevant stuff looked over. I'll circle back around to the tests as needed once I've got all the feature work done.

File: core/src/components/checkbox/test/basic/checkbox.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    await page.goto(`/src/components/checkbox/test/basic`);
+
+    const content = page.locator('#checkboxes');
+    expect(await content.screenshot()).toMatchSnapshot(`checkbox-basic-${page.getSnapshotSettings()}.png`);
+  });
+});
+
+test.describe('checkbox: ionChange', () => {
+  test.beforeEach(({ skip }) => {
+    skip.rtl();
```
Comment (liamdebeasi) [inline core/src/components/checkbox/test/basic/checkbox.e2e.ts:15]: Can we disable per-mode checks?

File: core/src/components/checkbox/test/indeterminate/checkbox.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { expect } from '@playwright/test';
+import { test } from '@utils/test/playwright';
+
+test.describe('checkbox: indeterminate', () => {
+  test('should not have visual regressions', async ({ page }) => {
```
Comment (liamdebeasi) [inline core/src/components/checkbox/test/indeterminate/checkbox.e2e.ts:5]: Does indeterminate state have RTL-specific logic? If not, we can probably disable the rtl check

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Noticed that some legacy screenshots got deleted:

`core/src/components/checkbox/test/legacy/basic/checkbox.e2e.ts-snapshots/checkbox-basic-md-ltr-Mobile-Safari-linux.png`
`core/src/components/checkbox/test/legacy/basic/checkbox.e2e.ts-snapshots/checkbox-basic-md-rtl-Mobile-Safari-linux.png`

This is due to the change in filename change in the basic legacy test, right?

I also am seeing a handful of non-checkbox screenshot diffs (datetime, picker-internal, textarea). Can't quite figure out what's different, so maybe a subtle difference?

**Review by liamdebeasi — APPROVED** [review]
Good to go once previous comments are addressed

**Comment by amandaejohnston** [comment]
> Noticed that some legacy screenshots got deleted:
> 
> `core/src/components/checkbox/test/legacy/basic/checkbox.e2e.ts-snapshots/checkbox-basic-md-ltr-Mobile-Safari-linux.png` `core/src/components/checkbox/test/legacy/basic/checkbox.e2e.ts-snapshots/checkbox-basic-md-rtl-Mobile-Safari-linux.png`
> 
> This is due to the change in filename change in the basic legacy test, right?

Yeah, those screenshots registered as removed and re-added rather than just renamed for some reason. The legacy screenshots that the diff is showing as net-new are the same as the ones that were "removed."

**Review by sean-perkins — APPROVED** [review]

**Comment by amandaejohnston** [comment]
Removed the new CSS properties and some extra code related to them; re-requesting review just in case 👀 I left the background props renamed to avoid confusion, since `--background` still sounds like it should be the background for the whole container.

**Review by liamdebeasi — APPROVED** [review]

**Comment by amandaejohnston** [comment]
Ah, looks like those failing tests are actually screenshot regressions 🤔 Gonna run the update job and make sure nothing goes awry there.

**Comment by amandaejohnston** [comment]
Okay, should be good now 👍

**Review by sean-perkins — APPROVED** [review]
Clicked through the examples, things look good to me 👍

**Pull request merged**